### DISCUSSION TITLE
.NET Resizing output does not match CLI Resizing

### DISCUSSION THREAD

**Discussion by spelech** [dlemstra/Magick.NET#1311] [Help] [answered]
I am using ImageMagick for resizing and creating PNG8s from large batches of source PNGs, TIFS and SVGs. We have several images that have some fine details (such as a long wire brush) that the usual calls to `Resize` and `AdaptiveResize` do not do well with.  They result in the wire brush head basically looking like Charlie Brown's Christmas tree.

Our PS method used their `Bicubic Sharper` resize method, and rather than implement a Bicubic interpolator myself, I am using Catrom as the filter type.  Here are the CLI arguments and then a snippet of .NET code.  The CLI output for resizing is what we want as it preserves the handle and wire brush head much better.  I am not converting to PNG8 in these examples, but this difference is not related to PNG8 conversion

`magick source.png -trim -filter Catrom -resize 300x100 PNG32:catrom_resize.png`

This is the .NET code we are using.  In our usage, we receive the png as a `byte[]` and have Magick.NET create an object with that.
`MagickImage srcImg = new MagickImage(_sourceImageBytes);`           
`srcImg.Trim();`
`srcImg.InterpolativeResize(300,0,PixelInterpolateMethod.Catrom);` 
`srcImg.Write(Path.Combine(savePath,fileName),MagickFormat.PNG32); `

I have also tried setting both srcImg.Interpolate and srcImg.Filter to Catrom and the results are the same.  Is there something we are missing with how the CLI loads an image versus how .NET does?

**Comment by dlemstra** [comment, marked as answer]
You will need to set `srcImg.FilterType` and call `.Resize` instead. I might need to improve this api in the future.

**Reply by spelech** [reply]
Thanks! If I might offer a suggestion, maybe overload the `Resize` function to include a filter type method so that it aligns with the CLI.  That way we can specify the Filter for that one operation rather than having to change the property for different calls.  A simple comment on the API saying it will use the `FilterType` property would also be a good addition.

Out of curiosity, when is `srcImg.Interpolate` used?  `InterpolativeResize` requires a method input so I wasn't sure when that property was used.

**Reply by dlemstra** [reply]
`Interpolate` is used in some of the internal methods but it is also an explicit argument for some of the methods. I also don't think that I can change the api for the `FilterType` property.
